pmd-core/src/main/java/net/sourceforge/pmd/cpd/TokenEntry.java

```diff
@@ -95,20 +95,22 @@ public static void clearImages() {
      */
     public static class State {
         private int tokenCount;
-        private Map<String, Integer> tokens;
-        private List<TokenEntry> entries;
+        private int tokensMapSize;
 
-        public State(List<TokenEntry> entries) {
+        public State() {
```

Comment:
@jsotuyod I wonder, how strict we should see compatibility here. Technically this changes TokenEntry.State, which is not internal, but public API and could potentially be used by someone...

Should we simply re-add these old constructor and method as deprecated?

I had a short look at CPD, and the TokenEntry class is part of the public API when retrieving the results of CPD (via Match -> Mark -> TokenEntry). So we can't simply internalize the whole TokenEntry class...

Reply:
good point. Yeah, that should work

Reply:
Probably, TokenEntry.State should be internal and deprecated